### PR TITLE
[WIP] refactor: Alert with hooks

### PR DESCRIPTION
**Pull request by orzyyyy**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | refactor Alert with hooks  |
| 🇨🇳 Chinese |     使用 hooks 重构 Alert    |

### ☑️ Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

File: components/alert/demo/banner.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ Display Alert as a banner at top of page.
 import { Alert } from 'antd';
 
 ReactDOM.render(
-  <div>
+  <>
```
Comment (dengfuping) [inline components/alert/demo/banner.md]: - There is no need to modify demo like that. It's better to make single PR minimal.

**Comment by zombieJ** [comment]
Could you help to use `rc-animate/lib/CSSMotion` instead of `Animation` from `rc-animate`?  Since `Animate` does not work well with function component.

**Comment by zombieJ** [comment]
And not modify current demo code since it will makes `master` merge `4.0-prepare` much potential conflict which we have already face to : )

**Comment by orzyyyy** [comment]
```jsx
<CSSMotion visible={closing} motionName={`${prefixCls}-slide-up`}>
  {() => (
    <div
      data-show={closing}
      className={alertCls}
      style={style}
      ref={wrapperRef}
      {...dataOrAriaProps}
    >
      {showIcon ? iconNode : null}
      <span className={`${prefixCls}-message`}>{message}</span>
      <span className={`${prefixCls}-description`}>{description}</span>
      {closeIcon}
    </div>
  )}
</CSSMotion>
```
Replace `Animate` with `CSSMotion`, and I ran into a problem:
How to bind `onEnd`?

I have checked props [there](https://github.com/react-component/animate/blob/master/src/CSSMotion.js#L32), but there seems to be no suitable hook.

File: components/alert/index.tsx [hunk 2]
```diff
@@ -64,47 +62,48 @@ const iconMapOutlined = {
...
 };
 
-export default class Alert extends React.Component<AlertProps, AlertState> {
-  state = {
-    closing: true,
-    closed: false,
-  };
-
-  handleClose = (e: React.MouseEvent<HTMLAnchorElement>) => {
+const Alert = (props: AlertProps) => {
```
Comment (yesmeck) [inline components/alert/index.tsx]:
```suggestion
const Alert: React.FC<AlertProps> = (props) => {
```

**Comment by orzyyyy** [comment]
>Since `Animate` does not work well with function component.

@zombieJ What are the side effects of `Animate`? Maybe I can help.

**Comment by orzyyyy** [comment]
Maybe it's not a right time to refactor now, close this.

**Pull request closed without merging**